coalib/settings/SectionFilling.py

```diff
@@ -60,6 +52,96 @@ def fill_section(section, acquire_settings, log_printer, bears):
     return section
 
 
+def get_required_and_optional_settings(bears):
+    """
+    Given a list of bears, retrieves the required and optional settings for
+    each bear, and returns as a dict.
+
+    :param bears: All bear classes or instances.
+    :return:      A tuple with a dict for required and a dict for optional
+                  settings. Each dict of settings has keys representing the
+                  setting name, and the value is a list of bear names for
+                  which that setting is required or optional.
+    """
+    required_settings = {}
+    optional_settings = {}
+    for bear in bears:
+        needed = bear.get_non_optional_settings()
+        optional = bear.get_optional_settings()
+        for key in needed:
+            if key in required_settings:
+                required_settings[key].append(bear.name)
+            else:
+                required_settings[key] = [needed[key][0],
+                                          bear.name]
+        for key in optional:
+            if key in optional_settings:
+                optional_settings[key].append(bear.name)
+            else:
+                optional_settings[key] = [optional[key][0],
+                                          bear.name]
+    return (required_settings, optional_settings)
+
+
+def warn_extraneous_settings(bears, parsed_settings):
+    """
+    Warns the user if any of the settings are not used by any of the
+    given bears.
+
+    :param bears:           All bear classes or instances.
+    :param parsed_settings: List of dicts, where each dict has a setting
+                            and value key, corresponding to each custom
+                            setting.
+    """
+    (required_settings,
+        optional_settings) = get_required_and_optional_settings(bears)
+    for setting_definition in parsed_settings:
+        if (setting_definition['setting'] not in required_settings and
+                setting_definition['setting'] not in optional_settings):
+            logging.warning('Setting \'{}\' is not used by any bear, '
```

Comment:
We need a test which captures the logging and shows this message was shown to the user.